src/core/load-fragment.c

```diff
@@ -5711,6 +5711,75 @@ int config_parse_cgroup_socket_bind(
         return 0;
 }
 
+int config_parse_restrict_network_interfaces(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        CGroupContext *c = data;
+        bool is_allow_rule = true;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->restrict_network_interfaces = set_free(c->restrict_network_interfaces);
+                return 0;
+        }
+
+        if (rvalue[0] == '~') {
+                is_allow_rule = false;
+                rvalue++;
+        }
+
+        if (!c->restrict_network_interfaces) {
+                c->restrict_network_interfaces = set_new(&string_hash_ops_free);
+                if (!c->restrict_network_interfaces)
+                        return log_oom();
```

Comment:
you can drop this explicit allocation. nowadays set_put_strdup() does that implicitly if needed, i.e. lazily